lib/ddq-backend-mysql.js

```diff
@@ -0,0 +1,374 @@
+"use strict";
+
+const EMIT_DATA = "data", EMIT_ERR = "error";
+
+module.exports = (configValidation, crypto, EventEmitter, mysql, timers) => {
+    var cycleCounter, writeRecord;
+
+    /**
+     * Updates heartbeatDate of a specific record and calls the callback that is
+     * passed in.
+     *
+     * @param {Object} ddqBackendInstance
+     * @param {string} recordId
+     * @param {Function} callback
+     */
+    function heartbeat(ddqBackendInstance, recordId, callback) {
+        ddqBackendInstance.connection.query(`UPDATE ??
+            SET heartbeatDate = NOW()
+            WHERE hash = ?
+                AND isProcessing = true
+                AND owner = ?;`,
```

Comment:
Working on the SQL guide right now. I'll post a link in Hipchat for everyone to look at when it's ready.

Good question; no semicolon is necessary. Removed here and in other cases.